### DISCUSSION TITLE
Wayland support?

### DISCUSSION THREAD

**Discussion by Newbytee** [trunkmaster/nextspace#382] [Q&A]
Are there plans/considerations to implement this as a Wayland compositor (could be done using, for example, [wlroots](https://github.com/swaywm/wlroots)) and/or port the utility applications to native Wayland?

**Comment by trunkmaster** [comment]
Nice question and nice point about wlroots! Sure I consider using Wayland in the future. 
Some actions were already taken by GNUstep project down this way. 

I hope I reach that point where Wayland support will go into my current TODO list...
Do you have some ideas, code prototype I can look into? I'm not familiar with Wayland yet...

**Comment by alexmyczko** [comment]
GNUstep does have a Wayland backend.
https://lists.gnu.org/archive/cgi-bin/namazu.cgi?query=wayland&submit=Search%21&idxname=gnustep-dev&max=20&result=normal&sort=score

**Comment by Newbytee** [comment]
> I hope I reach that point where Wayland support will go into my current TODO list...
> Do you have some ideas, code prototype I can look into? I'm not familiar with Wayland yet...

You could look into Weston, the reference implementation of Wayland: https://gitlab.freedesktop.org/wayland/weston/

There's also this introduction using wlroots to write Wayland compositors (however, somewhat outdated): https://drewdevault.com/2018/02/17/Writing-a-Wayland-compositor-1.html

Continuing the route of wlroots, there's tinywl: https://github.com/swaywm/wlroots/tree/master/tinywl

**Comment by alexmyczko** [comment]
it already exists!!! https://github.com/gnustep/libs-back/pull/17/commits/696ab6353488c203f4db68f1d62c9fd437bc0a6f#diff-659b64781af0473e6aa3f84a89e9ef8eR760

**Comment by andyarvanitis** [comment]
Looks like they have merged Wayland changes into master recently (though it looks like it's a WIP): https://github.com/gnustep/libs-back/pull/17

**Comment by zeroheure** [comment]
It is far from being ready. Code was merged "as is", in the hope that someone will take it and push it further.

**Comment by alexmyczko** [comment]
There is also https://github.com/phkaeser/wlmaker
